doc/contributing.rst

```diff
@@ -58,20 +60,46 @@ forward:
    In |NimSkull| repo, `testament`:cmd: (see below) runs all
    ``$nim/tests/*/t*.nim`` test files
 
+   - Usage of ``echo`` in tests is discouraged and should be replaced with
+     ``doAssert`` when possible. Assertion-based tests are reasier to read
+     (condition for test pass is written in the test itself) and move
+     around when needed (for running single piece of code for debuggin, or
+     moving whole test into another file)
+
+   - Don't forget to add `description:` field to the testament specification
+   - Also comment on the test's parts. Well-written test with careful
+     explanation of what is going on, or why particular element is tested
+     in this manner can serve as a powerful teaching tool.
+
+   -
+
 2. `runnableExamples` documentation comment tests, ran by `nim doc mymod.nim`:cmd:
    These end up in documentation and ensure documentation stays in sync with code.
 
-3. (deprecated) tests in `when isMainModule:` block, ran by `nim r mymod.nim`:cmd:.
+.. note:: Not all the tests follow the convention here, feel free to change
+          the ones that don't. Always leave the code cleaner than you found
+          it. Usage of deprecated tests styles is almost guaranteed to lead
+          to PR rejection.
+
+.. note:: Progress of the test suite improvements is tracked in the github
+          `project <https://github.com/nim-works/nimskull/projects/2>`_.
```

Comment:
I don't like the fact documentation is scattered all over the place for how specification should be written - I can agree that it can split into `contributing.rst` and `intern.rst`, but jumping back and forth with GitHub projects is something I would like to avoid if possible.

Reply:
Maybe I should just make `spec.rst`, there is a lot of text already

Reply:
Added `spec.rst` with explanation from the github project, when this PR is merged I will simplify project description.